### ISSUE TITLE
eth_estimateGas do not simulate transaction (and return revert reason)

### ISSUE THREAD

**Issue by BjMrq** [hashgraph/hedera-json-rpc-relay#493]
### Description

I deployed a simple contract that has a method that has a simple require that revert or not depending of a state value of the contract, to test revert behaviors.
For other EVM compatible networks estimateGas both estimat the gas needed for the transaction but also to check if the transaction will revert or not and if so what will be the revert condition.

This is an example of the result calling estimateGas on executing the method that is set to revert on the contract on Goerli.
```
(reason="execution reverted: Set to revert", method="estimateGas", transaction={"from":"0x1bCa0bf5e8796e45F4F25927AaB9BF2991fd3a5B","gasPrice":{"type":"BigNumber","hex":"0x0fe502a4"},"to":"0x1b68bA912B7AC0Cf771F36ac356Bbceda299fbcf","data":"0xe3a07ac0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000074368616e67656400000000000000000000000000000000000000000000000000","accessList":null}, error={"code":-32603,"message":"execution reverted: Set to revert","data":{"originalError":{"code":3,"data":"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d53657420746f2072657665727400000000000000000000000000000000000000","message":"execution reverted: Set to revert"}}}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.6.8)
```
Goerli contract: https://goerli.etherscan.io/address/0x1b68bA912B7AC0Cf771F36ac356Bbceda299fbcf
Hedera testnet contract: https://hashscan.io/#/testnet/contract/0.0.47958918

### Steps to reproduce

- deploy a simple contract that has a method with a simple require that revert
- call eth_estimateGas for this method on the deployed contract

### Hedera network

testnet

### Version

tesnet

**Comment by Nana-EC** [comment]
HI @BjMrq 

Thanks.
Similar to the other ticket an example contract and the call you're making will help to ensure we understand your exact flow.

However, similar to the other ticket our current flow is not capturing the revert reason and returning it.

There are 2 points to this.
- for queries the relay currently does not have access to the revert reason. This is something we're aware of and looking to address but is part of a more extensive effort under traceability. We'll brainstorm and see if there's anything to be done in the immediate 1-2 sprints
- for transactions the revert reasons for transaction that go through consensus do make it to the mirror node. So we'll look to add logic on the relay to retrieve the details there

**Comment by Nana-EC** [comment]
sdk-js has released https://github.com/hashgraph/hedera-sdk-js/releases/tag/v2.18.3 which should expose ContractFunctionResult.
Let's verify ti provides the needed info

**Comment by Ivo-Yankov** [comment]
Hi @BjMrq, this [PR](https://github.com/hashgraph/hedera-json-rpc-relay/pull/604) improves revert tracabillity. Both `eth_call` and `eth_getTransactionReceipt` will now return the revert reason, but unfortunatelly that is not possible for `eth_estimateGas` at this time.

**Comment by Nana-EC** [comment]
HI @BjMrq ,
As Ivo noted the transaction submissions will now expose the revert if applicable.
With the current implementation revert reasons for `eth_estimateGas` aren't possible.

In the future we do have plans to support Archive node operations.
Feel free to review the pending [HIP 584](https://github.com/hashgraph/hedera-improvement-proposal/pull/584) that would touch on this and ensure it addresses your desire and experience from other RPCs.

Moving this out of the current sprint backlog but leaving open for future work

**Comment by Nana-EC** [comment]
@BjMrq update on this ticket. We're making progress on the ability to return revert reasons on estimateGas.
We should have that resolved once https://github.com/hashgraph/hedera-json-rpc-relay/pull/1424 is merged. 
This is tentative for 0.27.0 of the relay.

**Comment by ebadiere** [comment]
PR #1510 fixed this and was released in 0.27.1.
